src/cli/commands/lint.ts

```diff
@@ -50,16 +51,32 @@ linting ./openapi.yaml
       char: 'v',
       description: 'increase verbosity',
     }),
+    config: flagHelpers.string({
+      char: 'c',
+      description: 'path to a config file',
+    }),
   };
 
   public static args = [{ name: 'source' }];
 
   public async run() {
     const { args, flags } = this.parse(Lint);
+    const { config: configFileFlag } = flags;
+    let config: IConfig = mergeConfig(createEmptyConfig(), flags);
+
+    const configFile = configFileFlag || getDefaultConfigFile(process.cwd()) || null;
```

Comment:
I'd say we should follow ESLint / TSLint and general ecosystem (i.e. Babel) a bit more here.
tl;dr

- package.json field, i.e. `spectral`
- `.spectralrc`, `.spectral.yaml`, `.spectral.json`
- js files should be supported for easier merging - environment based configuration is way nicer to implement
- file relative configuration is optional, we don't need it now

https://eslint.org/docs/2.0.0/user-guide/configuring#using-configuration-files
https://babeljs.io/docs/en/config-files#configuration-file-types

@philsturgeon thoughts?

Reply:
We don't want to rely on package.json as this is usable via a binary (not always installed via npm or yarn).

We can only use .spectralrc if its some special file type, so lets stick with `.spectral.yaml` and `.spectral.json`. 

"js files should be supported for easier merging" What do you mean about merging? There should not be any merging in config files.

Reply:
> We don't want to rely on package.json as this is usable via a binary (not always installed via npm or yarn).

Yeah, that's why it'd be optional. There are plenty of projects that tend to configure tools using package.json, so this would be nice to have, but we can skip it for now.

> "js files should be supported for easier merging" What do you mean about merging? There should not be any merging in config files.

True, I wasn't clear enough.
In our case, it would be something like

```ts
verbose: !!process.env.CI
```

Nothing crucial at the very moment, since our configuration isn't very complex yet, but that would be a nice addition in the future once we add more options.

Reply:
Gotcha oin the package stuff, but right now one way is ok, don't need to make two ways. We are just making a config file.

I think that JS way might be overthinking things a bit, people can combine a config file with CLI options, they would just use `lint --verbose` in their CI script.

Reply:
Sounds good!
In general, this is more food for thoughts. Our configuration is still fairly limited, so we can simply revisit the above ideas if we need to make it more robust one day.

Reply:
I'm happy with keeping it simple for now, one default config file path should be enough for this story, we can add support to other paths later.